dashboard/app/coverage.go

```diff
@@ -56,22 +56,94 @@ type funcStyleBodyJS func(
 	scope *coveragedb.SelectScope, onlyUnique bool, sss, managers []string, dataFilters cover.Format,
 ) (template.CSS, template.HTML, template.HTML, error)
 
+type coverageHeatmapParams struct {
+	manager    string
+	subsystem  string
+	onlyUnique bool
+	periodType string
+	nPeriods   int
+	dateTo     civil.Date
+	cover.Format
+}
+
+func makeHeatmapParams(ctx context.Context, r *http.Request) (*coverageHeatmapParams, error) {
+	// Prefixing "0" we don't fail on empty string.
+	minCoverLinesDrop, err := strconv.Atoi("0" + r.FormValue(covPageParams[MinCoverLinesDrop]))
```

Comment:
I think this code would benefit from a helper that extracts an integer parameter or returns a default value: `getHttpFormIntValue(r, covPageParams[MinCoverLinesDrop], /* default */ 0)`, or `getHttpFormIntValue(r, covPageParams[PeriodCount], /* default */ 4)`.